### PR TITLE
[1.13] Hook up new dcos-diagnostics API

### PR DESCRIPTION
**Pull request by janisz**
Fixes: https://jira.mesosphere.com/browse/DCOS_OSS-5447
    
Backports:

* https://github.com/dcos/dcos-core-cli/pull/255 (only Dockerfile and Makefile)
* https://github.com/dcos/dcos-core-cli/pull/335
* https://github.com/dcos/dcos-core-cli/pull/329
* https://github.com/dcos/dcos-core-cli/pull/326
* https://github.com/dcos/dcos-core-cli/pull/320
* https://github.com/dcos/dcos-core-cli/pull/317
* https://github.com/dcos/dcos-core-cli/pull/340
* https://github.com/dcos/dcos-core-cli/pull/339
* https://github.com/dcos/dcos-core-cli/pull/335
* https://github.com/dcos/dcos-core-cli/pull/336
* https://github.com/dcos/dcos-core-cli/pull/333
* https://github.com/dcos/dcos-core-cli/pull/329
* https://github.com/dcos/dcos-core-cli/pull/326
* https://github.com/dcos/dcos-core-cli/pull/323
* https://github.com/dcos/dcos-core-cli/pull/317

### CODE REVIEW

File: pkg/cmd/node/node_diagnostics_delete.go [hunk 1]
```diff
@@ -13,6 +13,11 @@ func newCmdNodeDiagnosticsDelete(ctx api.Context) *cobra.Command {
 		Short: "Delete a diagnostics bundle",
 		Args:  cobra.ExactArgs(1),
 		RunE: func(cmd *cobra.Command, args []string) error {
+			err := ctx.Deprecated("This command is deprecated since DC/OS 1.14, please use 'dcos diagnostics delete' instead.")
```
Comment (br-lewis) [inline pkg/cmd/node/node_diagnostics_delete.go:16]: Deprecation is for the wrong version but I'm also not sure we'll want to backport the deprecation warning.

**Comment by br-lewis** [comment]
I could be missing it but I don't see any changes to [`dcos.go`](https://github.com/dcos/dcos-core-cli/blob/1.13-patch.x/pkg/cmd/dcos.go#L21-L24) to add the new diagnostics command to the root command.

**Comment by janisz** [comment]
@mesosphere-ci run linux integration tests

**Comment by janisz** [comment]
@br-lewis Good catch, fixed. Thnks

**Comment by janisz** [comment]
@mesosphere-ci run linux integration tests

**Comment by bamarni** [comment]
@janisz : ok now I understand your concerns from yesterday about the pain of backporting with cherry-picks :sweat_smile: 

As the diagnostic work was done in multiple PRs and is not finished yet, I think it would have been easier to wait for it to be finished in the 1.14 branch, then backport by copying the new command and client folders into a single new commit.

**Comment by janisz** [comment]
@bamarni I did it exactly as you advise. 
```
git co 1.14-patch.x completion/bash/diagnostics.sh 
git co 1.14-patch.x pkg/diagnostics/v2/ 
git co 1.14-patch.x pkg/cmd/diagnostics/v2  
```

But makefile and dockerfile reuqired manual fixes

**Comment by janisz** [comment]
> [We have to discuss what we're going to do with this ticket since backporting the CLI seems to not be as easy as we thought it would be.
](https://jira.mesosphere.com/browse/DCOS_OSS-5447?focusedCommentId=295304&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-295304)

**Pull request closed without merging**